### PR TITLE
Improve default context component

### PR DESCRIPTION
**Pull request by aaranadev**
A little refactor of the current components improve the context and lifecycle to a better performance

### CODE REVIEW

File: template-skeleton/template/src/components/common/Map.js [hunk 1]
```diff
@@ -6,6 +6,11 @@ import { makeStyles, useTheme, useMediaQuery } from '@material-ui/core';
 import { setViewState } from '@carto/react-redux';
 import { BASEMAPS, GoogleMap } from '@carto/react-basemaps';
 
+const MAPS_TYPES = {
```
Comment (borja-munoz) [inline template-skeleton/template/src/components/common/Map.js:9]: Maybe BASEMAP_TYPES, instead of MAPS_TYPES

Comment (aaranadev) [reply template-skeleton/template/src/components/common/Map.js]: Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 - Update to latest carto-react (v1.1.0-alpha.3) adapting to new SQL API (executeSQL) [#242](https://github.com/CartoDB/carto-react-template/pull/242)
 - Update to latest deck.gl (8.5.0-alpha.10) [#242](https://github.com/CartoDB/carto-react-template/pull/242)
 - Allow public apps with Cloud Native [#245](https://github.com/CartoDB/carto-react-template/pull/245)
+- Improve code with props destructuring and components with context (atomic design) [#247](https://github.com/CartoDB/carto-react-template/pull/247)
```
Comment (VictorVelarde) [inline CHANGELOG.md:14]: Wrong PR id, right? This is number 250

File: template-skeleton/template/src/components/common/Map.js [hunk 3]
```diff
@@ -114,10 +117,14 @@ function Map({ layers }) {
...
       />
-    );
-  }
+    ),
+  };
 
-  return <div className={classes.root}>{map}</div>;
-}
+  let map = mapsAvailable[basemap.type] ? (
+    mapsAvailable[basemap.type]()
```
Comment (VictorVelarde) [inline template-skeleton/template/src/components/common/Map.js:124]: 👍🏻

**Review by VictorVelarde — CHANGES_REQUESTED** [review]
Overall it looks good, but please bring master in and ping me back. I did a try, but some merges / rebase operations were not obvious, so I would please ask you to update it

**Comment by aaranadev** [comment]
@VictorVelarde the chantes requested are done

**Review by VictorVelarde — APPROVED** [review]
Nice

**Pull request merged**